packages/shared/src/UI/components/I18NextProviderHMR.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect, useState } from 'react'
+import { I18nextProvider, type I18nextProviderProps } from 'react-i18next'
+
+export function I18NextProviderHMR({ i18n, defaultNS, children }: React.PropsWithChildren<I18nextProviderProps>) {
```

Comment:
I would suggest:

```ts
 export const I18NextProvider = process.env.NODE_ENV === 'development' ? I18NextProviderHMR : I18NextProviderWithoutHMR
```
Then you don't need to modify `<I18NextProvider />` in those node tree in above files. After all, they don't need to care about is the `<I18NextProvider />` hot-reload-able or not, do they?